week3/project/ecommerce/src/hooks/useFetch.jsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react'
+
+export default function useFetch() {
+   const [data, setData] = useState([]);
+   const [isLoading, setIsLoading] = useState(false);
+   const [errorObj, setErrorObj] = useState({ isError: false, message: '' })
+
+   async function getFetch(url) {
+      try {
+         setIsLoading(true);
+         setErrorObj({ isError: false, message: '' })
+         const response = await fetch(url);
+         const data = await response.json();
+         // return data;
+         setData(data);
+      }
+      catch (error) {
+         setErrorObj({ isError: true, message: error.message })
+      }
+      finally {
+         setIsLoading(false);
+      }
+   }
+
+   return ({
+      data,
+      isLoading,
+      errorObj,
+      getFetch
+   })
```

Comment:
you can return objects without using round brackets 
```js
return { data, isLoading, errorObj, getFetch };
```

but when you use the arrow function and with one-line syntax then it's important to use round brackets.
```js
() => ({...whatever})
```
📝advise📝